src/main/java/com/bookshop/controller/AuthController.java

```diff
@@ -0,0 +1,44 @@
+package com.bookshop.controller;
+
+import com.bookshop.dto.user.UserLoginRequestDto;
+import com.bookshop.dto.user.UserLoginResponseDto;
+import com.bookshop.dto.user.UserRegistrationRequestDto;
+import com.bookshop.dto.user.UserResponseDto;
+import com.bookshop.security.AuthenticationService;
+import com.bookshop.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/auth")
+@Tag(name = "Authentication controller",
+        description = "Endpoints for login and registration")
+public class AuthController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @Operation(summary = "Register",
+                description = "Register new user")
+    @PostMapping("/register")
+    @ResponseStatus(HttpStatus.CREATED)
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request) {
+        return userService.register(request);
+    }
+
+    @Operation(summary = "Login",
+                description = "Login user by their credentials")
+    @PostMapping("/login")
+    @ResponseStatus(HttpStatus.OK)
```

Comment:
just FYI

`OK` is the default response status, so you can omit it

remember about this in future :)